src/parser/whamm_parser.rs

```diff
@@ -576,6 +566,36 @@ fn expr_from_pairs(pairs: Pairs<Rule>) -> Result<Expr, Vec<WhammError>> {
         .map_primary(|primary| -> Result<Expr, Vec<WhammError>> {
             expr_primary(primary)
         })
+        .map_prefix(|op, rhs|  -> Result<Expr, Vec<WhammError>> {
+            return match rhs {
+                Ok(rhs) => {
+                    let op = match op.as_rule() {
+                        Rule::neg => Op::Neg,
+                        rule => {
+                            return Err(vec![ErrorGen::get_parse_error(
+                                true,
+                                Some(UNEXPECTED_ERR_MSG.to_string()),
+                                Some(LineColLocation::from(op.as_span())),
+                                vec![Rule::prefix],
+                                vec![rule])]);
+                        },
+                    };
+
+                    let rhs_line_col = if let Some(rhs_loc) = rhs.loc() {
+                        rhs_loc.line_col.clone()
+                    } else {
+                        exit(1);
```

Comment:
Instead of exiting, this should return a fatal error (like above). If the expression doesn't have a location associated with it, the compiler went wrong somewhere.